modelling/src/main/java/org/axonframework/modelling/configuration/DefaultStateBasedEntityBuilder.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.modelling.configuration;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.configuration.ComponentFactory;
+import org.axonframework.modelling.SimpleRepository;
+import org.axonframework.modelling.SimpleRepositoryEntityLoader;
+import org.axonframework.modelling.SimpleRepositoryEntityPersister;
+import org.axonframework.modelling.repository.AsyncRepository;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Basis implementation of the {@link StateBasedEntityBuilder}.
+ *
+ * @param <I> The type of identifier used to identify the state-based entity that's being built.
+ * @param <E> The type of the state-based entity being built.
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+class DefaultStateBasedEntityBuilder<I, E> implements
```

Comment:
I do like the `<I,E>` here, I think we should align other classes to this in the future.

Reply:
Awesome, then let's!